query/src/users/auth/auth_mgr.rs

```diff
@@ -85,6 +98,15 @@ impl AuthMgr {
                 password: p,
                 hostname: h,
             } => {
+                let identity = UserIdentity {
+                    username: n.to_string(),
+                    hostname: h.as_ref().unwrap_or(&"%".to_string()).to_string(),
+                };
+                if identity.is_root() && !identity.is_localhost() {
+                    return Err(ErrorCode::AuthenticateFailure(
+                        "root user can only login from localhost",
+                    ));
+                }
```

Comment:
I think we should check it on https://github.com/datafuselabs/databend/blob/main/query/src/users/user_mgr.rs#L31-L41

Reply:
`user_mgr:get_user` method will also be called by `ShowGrantsInterpreter` and `DfAlterUser`, and these two seems to be irrelevant to auth logic so I put the host check in `auth_mgr`.

Reply:
Why not return unknown user directly in get_user?